### PR TITLE
Refactor ResponseParser.h and initial ResponseParser implement without unit test

### PR DESCRIPTION
**Pull request by LinRenHong**

### CODE REVIEW

File: src/Implement/ResponseParser.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ResponseParser.h"
+#include "OperateData.h"
```
Comment (Jazz86) [inline src/Implement/ResponseParser.cpp:6]: clang-format.

File: src/Implement/ResponseParser.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        size_t value_start = response.find('"', key_end + 1) + 1;
+        size_t value_end = response.find('"', value_start);
+        new_data->body[response.substr(key_start, key_end - key_start)] = response.substr(value_start, value_end - value_start);
+        iterate_index = value_end + 1;
+    }
+
+    return new_data;
+}
+
+ResponseParser::ResponseParser() {
```
Comment (Jazz86) [inline src/Implement/ResponseParser.cpp:74]: clang-format.

**Review by Jazz86 — COMMENTED** [review]
should fix the conflict first.

File: src/Implement/ResponseParser.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    // Get status code
+    size_t status_code_start = response.find(" ") + 1;
+    size_t status_code_end = response.find(" ", status_code_start);
+    new_data->header["status_code"] = response.substr(status_code_start, status_code_end - status_code_start);
+    
+    // Get status
+    new_data->header["status"] = response.substr(status_code_end + 1, response.find('\n') - status_code_end - 1);
+    
+    // Use Anonymous function to get header value
+    auto get_header_value_by_key = [] (std::string response, std::string header_key) {
```
Comment (john0312) [inline src/Implement/ResponseParser.cpp:42]: Consider using const ref to save the copying

File: src/Implement/ResponseParser.cpp [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        size_t header_value_start = response.find(header_key) + header_key.length() + 2;
```
Comment (john0312) [inline src/Implement/ResponseParser.cpp:43]: What if it's not found? Please do error handling.

File: src/Implement/ResponseParser.cpp [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    
+    // Get header value
+    new_data->header["Server"] = get_header_value_by_key(response, "Server");
+    new_data->header["Date"] = get_header_value_by_key(response, "Date");
+    new_data->header["Connection"] = get_header_value_by_key(response, "Connection");
+    new_data->header["Content-type"] = get_header_value_by_key(response, "Content-Type");
+    new_data->header["Content-length"] = get_header_value_by_key(response, "Content-Length");
+    
+    // Parse body
+    size_t body_start_pos = response.find("{");
```
Comment (john0312) [inline src/Implement/ResponseParser.cpp:56]: We might find '{' and '}' in the header too. It is possible to have '}' in the body as well.

File: src/Implement/ResponseParser.cpp [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    size_t body_end_pos = response.find("}");
+    size_t iterate_index = body_start_pos;
+    
+    // All the json key and value is string, so will be cut by finding "
+    // Each loop will get a key and a value
+    while (iterate_index < body_end_pos) {
+        size_t key_start = response.find('"', iterate_index) + 1;
+        size_t key_end = response.find('"', key_start);
+        size_t value_start = response.find('"', key_end + 1) + 1;
+        size_t value_end = response.find('"', value_start);
```
Comment (john0312) [inline src/Implement/ResponseParser.cpp:66]: Error handling when they are not found?

**Pull request closed without merging**